imagebuildah/stage_executor.go

```diff
@@ -363,24 +363,78 @@ func (s *StageExecutor) Copy(excludes []string, copies ...imagebuilder.Copy) err
 		preserveOwnership := false
 		contextDir := s.executor.contextDir
 		if len(copy.From) > 0 {
+			// If additionalContext is selected do not process anything else
+			didSelectAdditionalContext := false
+			// If additionalContext is selected but it was image only process
+			// image and ignore stages if any. [ Buildx parity ]
+			didSelectImageAsAdditionalContext := false
 			// If from has an argument within it, resolve it to its
 			// value.  Otherwise just return the value found.
 			from, fromErr := imagebuilder.ProcessWord(copy.From, s.stage.Builder.Arguments())
 			if fromErr != nil {
 				return errors.Wrapf(fromErr, "unable to resolve argument %q", copy.From)
 			}
-			if isStage, err := s.executor.waitForStage(s.ctx, from, s.stages[:s.index]); isStage && err != nil {
-				return err
+			if additionalBuildContext, ok := s.executor.additionalBuildContexts[from]; ok {
+				if additionalBuildContext != nil {
+					didSelectAdditionalContext = true
+					if !additionalBuildContext.IsImage {
+						contextDir = additionalBuildContext.Value
+						if additionalBuildContext.IsURL {
+							// Check if following buildContext was already
+							// downloaded before in any other RUN step. If yes
+							// lets check if we are hitting any cache and cache
+							// actually exists on host.
+							cacheExistsOnHost := false
+							if additionalBuildContext.DownloadedCache != "" {
+								if stat, err := os.Stat(additionalBuildContext.DownloadedCache); err == nil && stat.IsDir() {
+									cacheExistsOnHost = true
+								}
+							}
```

Comment:
It looks like `cacheExistsOnHost` can be eliminated, since `DownloadedCache` is only ever set after the directory is created and populated.